### PR TITLE
Remove workerSpec from AdapterSpec

### PR DESCRIPTION
**Pull request by gokulavasan**
Removing WorkerSpec from AdapterSpec since it can be retrieved from the ApplicationSpec from Store. The only information that is required to be passed on from the Template's configureAdpater method is instances and runtime args (in case of Realtime Adapter) and ScheduleSpecification (in case of Batch Adapter) since we don't allow schedules to be part of Application Template.

More changes need to be done to AdapterSpec but they will be addressed in subsequent PRs so that PR #1962 can make use of these changes.

Bamboo : http://builds.cask.co/browse/CDAP-DUT1405-3

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/adapter/AdapterService.java [hunk 1]
```diff
@@ -344,14 +344,18 @@ public void startAdapter(Id.Namespace namespace, String adapterName)
     setAdapterStatus(namespace, adapterName, AdapterStatus.STARTED);
   }
 
+  private Id.Program getWorkflowId(Id.Namespace namespace, AdapterSpecification adapterSpec) {
+    ApplicationSpecification appSpec = store.getApplication(Id.Application.from(namespace, adapterSpec.getTemplate()));
+    Preconditions.checkNotNull(appSpec);
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/adapter/AdapterService.java]: Do you want to throw `NullPointerException` or something more meaningful like `NotFoundException`?

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/adapter/AdapterService.java [hunk 2]
```diff
@@ -344,14 +344,18 @@ public void startAdapter(Id.Namespace namespace, String adapterName)
...
+    Preconditions.checkArgument(appSpec.getWorkflows().size() == 1);
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/adapter/AdapterService.java]: Same comment as above.

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/adapter/AdapterService.java [hunk 3]
```diff
@@ -344,14 +344,18 @@ public void startAdapter(Id.Namespace namespace, String adapterName)
...
+    String workflowName = Lists.newArrayList(appSpec.getWorkflows().keySet()).get(0);
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/adapter/AdapterService.java]: Why need to new an array list? Use `Iterables.getFirst(appSpec.getWorkflows().keySet())`.

File: cdap-app-fabric/src/main/java/co/cask/cdap/templates/DefaultAdapterConfigurer.java [hunk 4]
```diff
@@ -165,33 +163,23 @@ public void addRuntimeArgument(String key, String value) {
   }
 
   public AdapterSpecification createSpecification() {
+    ScheduleSpecification scheduleSpec = null;
+    if (programType == ProgramType.WORKFLOW) {
+      String workflowName = templateSpec.getWorkflows().values().iterator().next().getName();
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/templates/DefaultAdapterConfigurer.java]: You can use `Iterables.getFirst`.

**Comment by chtyim** [comment]
Just couple minor comments. The rest LGTM.

**Comment by gokulavasan** [comment]
@chtyim Addressed comments. Please take a look again. Thanks!

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/adapter/AdapterService.java [hunk 5]
```diff
@@ -344,14 +345,23 @@ public void startAdapter(Id.Namespace namespace, String adapterName)
     setAdapterStatus(namespace, adapterName, AdapterStatus.STARTED);
   }
 
+  private Id.Program getWorkflowId(Id.Namespace namespace, AdapterSpecification adapterSpec) throws NotFoundException {
+    Id.Application appId = Id.Application.from(namespace, adapterSpec.getTemplate());
+    ApplicationSpecification appSpec = store.getApplication(appId);
+    try {
+      Preconditions.checkArgument(appSpec != null);
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/adapter/AdapterService.java]: maybe just a simple `if:

```java
if (appSpec == null || appSpec.getWorkflows().size() != 1) {
  throw new NotFoundException(appId);
}
```

**Comment by gokulavasan** [comment]
@chtyim Addressed comments. Please take a look. Thanks!

**Comment by chtyim** [comment]
LGTM

**Pull request merged**